src/uct/tcp/tcp_iface.c

```diff
@@ -231,8 +241,7 @@ static ucs_status_t uct_tcp_iface_query(uct_iface_h tl_iface,
         return status;
     }
 
-    ucs_snprintf_safe(sysfs_path, PATH_MAX, "%s/%s/device",
-                      UCT_TCP_IFACE_NETDEV_DIR, iface->if_name);
+    sysfs_path             = uct_tcp_iface_get_sysfs_path(iface->if_name, path_buffer);
```

Comment:
so the problem was that need to remove "/device" here?
won't it break anything? since seems this fix is not BF/arm specific

Reply:
Actually this was a bug, because we should not use /device at all.
All the logic for the 2 cases (x86/Arm) is covered in `uct_iface_get_sysfs_path`

Reply:
so it didn't work at all, even on x86?

Reply:
on x86 it worked by chance, because `device` file points exactly to the location of the required files (`current_link_width/speed`)

Reply:
seems that without device it wouldn't work on x86
```
13:35 <yosefe@jazz07:5 launch>$ ll /sys/class/net/ib0/current_link_speed
ls: cannot access /sys/class/net/ib0/current_link_speed: No such file or directory
13:35 <yosefe@jazz07:5 launch>$ ll /sys/class/net/ib0/device/current_link_speed
-r--r--r-- 1 root root 4096 Jan 15 13:35 /sys/class/net/ib0/device/current_link_speed
```

Reply:
please add some unit test that makes sure this process works

Reply:
we have the `select_transport_rma_bw` which failed before the fix and pass after it.
do we need another test?

Reply:
yeah, something that always fails if we cannot read the relevant sysfs file, regardless of lane selection
maybe check that sys_dev in uct_query_resources is not UNKNOWN

Reply:
There's no simple way to do this, because we need to check the value of pci_bw and compare it with something.
The 2 options I see are:
1) store sysfs_path as a new member field on tcp_iface, and then use it to cal pci_bw. (then we can check it's value from test).
2) call `uct_tcp_iface_get_sysfs_path` and `get_pci_bw` from test to get the expected value and compare it with the result of `uct_iface_query`)

Reply:
option 2: wrap uct_tcp_iface_get_sysfs_path+ get_pci_bw in some helper function in TCP transport, and call it explicitly from the test. this function can take tcp iface name and return its pci bw.

Reply:
@shasson5  please check whether this one is still relevant.